test/components/StatefulButton.test.tsx

```diff
@@ -108,9 +108,7 @@ describe('StatefulButton', () => {
       const buttonInner = component.find('.button-inner');
       buttonInner.simulate('click');
       expect(buttonInner.hasClass('is-processing')).toBe(true);
-      const newProps = clone(component.props);
-      newProps.baseState = 'not-allowed';
-      component.setProps(newProps);
+      component.setProps({ baseState: 'not-allowed' });
```

Comment:
not strictly related, but since I was touching `StatefulButton` I fixed this error.